src/tokens/saga.ts

```diff
@@ -259,14 +262,23 @@ export function* watchAccountFundedOrLiquidated() {
         const isAccountFundedBefore = prevTokenBalance?.gt(DOLLAR_MIN_AMOUNT_ACCOUNT_FUNDED)
         const isAccountFundedAfter = tokenBalance?.gt(DOLLAR_MIN_AMOUNT_ACCOUNT_FUNDED)
 
-        if (isAccountFundedBefore && !isAccountFundedAfter) {
+        if (
+          isAccountFundedBefore &&
+          !isAccountFundedAfter &&
+          prevNetworkIds.every((value) => supportedNetworkIds.includes(value))
```

Comment:
can you add a comment explaining this (not super obvious) and add unit tests for the different scenarios? Also consider using sets over arrays.